### PR TITLE
Changes the contract to make it simpler and rely on the realm available from the current session

### LINKED ISSUE #28403: Changes the contract to make it simpler and rely on the realm available from the current session
### Description

Changes the contract to make it simpler and rely on the realm available from the current session

### PR DESCRIPTION
**Pull request by pedroigor**
* Providers operating within the scope of a realm are always created based on a `KeycloakSession` from where the realm is already available so that there is no need to complicate APIs by having the `RealmModel` as a parameter to every single method.
*  This same pattern can be seem in different providers and provides a much cleaner contract for a `Provider` that only works within the scope of a realm

Closes #28403

### CODE REVIEW

**Review by vramik — CHANGES_REQUESTED** [review]
Thank you @pedroigor for preparing this PR. I'd suggest to keep the `RealmModel` as a parameter in the methods as it'd be consistent with other model related Providers in the codebase:

https://github.com/keycloak/keycloak/blob/main/server-spi/src/main/java/org/keycloak/models/ClientProvider.java
https://github.com/keycloak/keycloak/blob/main/server-spi/src/main/java/org/keycloak/models/ClientScopeProvider.java
https://github.com/keycloak/keycloak/blob/main/server-spi/src/main/java/org/keycloak/models/GroupProvider.java
https://github.com/keycloak/keycloak/blob/main/server-spi/src/main/java/org/keycloak/models/RoleProvider.java
https://github.com/keycloak/keycloak/blob/main/server-spi/src/main/java/org/keycloak/models/UserLoginFailureProvider.java
https://github.com/keycloak/keycloak/blob/main/server-spi/src/main/java/org/keycloak/models/UserProvider.java
https://github.com/keycloak/keycloak/blob/main/server-spi/src/main/java/org/keycloak/models/UserSessionProvider.java

and storage related examples

https://github.com/keycloak/keycloak/blob/main/server-spi/src/main/java/org/keycloak/storage/client/ClientLookupProvider.java
https://github.com/keycloak/keycloak/blob/main/server-spi/src/main/java/org/keycloak/storage/clientscope/ClientScopeLookupProvider.java
https://github.com/keycloak/keycloak/blob/main/server-spi/src/main/java/org/keycloak/storage/group/GroupLookupProvider.java
https://github.com/keycloak/keycloak/blob/main/server-spi/src/main/java/org/keycloak/storage/user/UserQueryMethodsProvider.java
https://github.com/keycloak/keycloak/blob/main/server-spi/src/main/java/org/keycloak/storage/user/UserCountMethodsProvider.java

I'm aware there are some providers, like https://github.com/keycloak/keycloak/blob/main/server-spi/src/main/java/org/keycloak/models/SingleUseObjectProvider.java not following the same pattern and it's not in scope atm to unify it. 

In summary I'd like to try to keep some unified approach there. If you aim to simplify the contracts of the methods, I belive it should be done consistently across other providers. That'd require some broader discussion and agreement.

Given said that I'd be in favor of not merging this PR.

File: model/jpa/src/main/java/org/keycloak/organization/jpa/JpaOrganizationProvider.java [hunk 1]
```diff
@@ -42,16 +42,21 @@ public class JpaOrganizationProvider implements OrganizationProvider {
...
     private final UserProvider userProvider;
+    private final RealmModel realm;
 
     public JpaOrganizationProvider(KeycloakSession session) {
         em = session.getProvider(JpaConnectionProvider.class).getEntityManager();
         groupProvider = session.groups();
         userProvider = session.users();
+        realm = session.getContext().getRealm();
+        if (realm == null) {
+            throw new IllegalArgumentException("Session not bound with a realm");
```
Comment (sguilhen) [inline model/jpa/src/main/java/org/keycloak/organization/jpa/JpaOrganizationProvider.java:53]:
```suggestion
            throw new IllegalArgumentException("Session not bound to a realm");
```

**Comment by pedroigor** [comment]
@vramik I see. I'm favoring simplicity over a consistency we don't have yet and requires changing other parts of the code base.

There is no need to have a method in an interface (`getRealmModel`) if we are not using it in any place (at least for now) just for the sake of consistency.

**Review by sguilhen — APPROVED** [review]

**Review by ahus1 — CHANGES_REQUESTED** [review]
I don't see a reference to an issue, and without such a reference it won't appear in the release notes.

Please create a related issue. The issue can be added to the commit when using the GitHub UI's squash functionality.

**Comment by pedroigor** [comment]
@ahus1 Added the issue.

**Review by ahus1 — APPROVED** [review]
Thank you! Approving as of @sguilhen's review.
